src/transactions.ts

```diff
@@ -21,18 +21,22 @@ export async function poolWithdrawBonds(pool: FungiblePool, signer: Signer) {
   try {
     const contractPoolWithSigner = pool.contract.connect(signer);
     const recipient = await signer.getAddress();
-    const nonce = await NonceTracker.getNonce(signer);
-    const tx = await withdrawBonds(
-      contractPoolWithSigner,
-      recipient,
-      MAX_UINT_256,
-      {
-        nonce: nonce.toString(),
-      }
-    );
-    await tx.verifyAndSubmit();
+    
+    // Use queueTransaction instead of manual nonce management
+    await NonceTracker.queueTransaction(signer, async (nonce) => {
+      const tx = await withdrawBonds(
+        contractPoolWithSigner,
+        recipient,
+        MAX_UINT_256,
+        {
+          nonce: nonce.toString(),
+        }
+      );
+      return await tx.verifyAndSubmit();
+    });
   } catch (error) {
-    NonceTracker.resetNonce(signer, address);
```

Comment:
Many of the catches in this file now have no purpose since they are simply rethrowing.  Better to remove the `try...catch` blocks entirely rather than rethrow.

Reply:
agree, removed all the try, catch blocks because errors are taken care of queueTransaction